src/macro-context.js

```diff
@@ -295,16 +306,47 @@ export default class MacroContext {
 
   _rest(enf) {
     const priv = privateData.get(this);
-    if(priv.backup === enf) {
+    if(priv.markers.get(priv.startMarker) === enf) {
       return priv.enf.rest;
     }
     throw Error("Unauthorized access!");
   }
 
-  reset() {
+  reset(marker) {
+    const priv = privateData.get(this);
+    let enf;
+    if(marker == null) {
```

Comment:
Turns out I did have my linter on. Do you want me to add the rule to `.eslintrc`?